xarray/core/dataset.py

```diff
@@ -4184,6 +4184,12 @@ def _validate_interp_indexer(x, new_x):
         variables: dict[Hashable, Variable] = {}
         reindex: bool = False
         for name, var in obj._variables.items():
+            if var.isnull().any():
```

Comment:
ah we can't do this really. I thought the behaviour stemmed from NaNs in the coordinate. Scanning the whole array like this is going to be expensive in many cases.